File: DSA/Problems/countDistsinctsBitwiseOr.js
function countDistsinctsBitwiseOr(arr) {
  let result = new Set();
  let prev = new Set();
  for (let x of arr) {
    let curr = new Set();
    curr.add(x);

    for (let val of prev) {
      curr.add(x | val);
    }

    for (let val of curr) {
      result.add(val);
    }
    prev = curr;
  }
  return result.size;
}

console.log(`Ans is ${countDistsinctsBitwiseOr([1, 2, 3])}`);

File: DSA/Problems/bitwise/countDistsinctsBitwiseOr.py
def countDistsinctsBitwiseOr (arr):
    result = set()
    prev = set()
    for x in arr:
        curr = set()
        curr.add(x)

        for val in prev:
            curr.add(val | x)

        for val in curr:
            result.add(val)

        prev = curr
    
    return len(result)

print(countDistsinctsBitwiseOr([1,2,3]))
